test/integration/targets/azure_rm_keyvault/tasks/main.yml

```diff
@@ -0,0 +1,114 @@
+- name: Create random keyvault name
+  set_fact:
+    keyvault_orig: "{{ 'ip' + resource_group | hash('md5') | truncate(16, True, '') + (65535 | random | string) }}"
+    keyvault_tags: "{{ 'ip' + resource_group | hash('md5') | truncate(16, True, '') + (64535 | random | string) }}"
+    keyvault_secr: "{{ 'ip' + resource_group | hash('md5') | truncate(16, True, '') + (63535 | random | string) }}"
+    keyvault_keys: "{{ 'ip' + resource_group | hash('md5') | truncate(16, True, '') + (62535 | random | string) }}"
+    keyvault_cert: "{{ 'ip' + resource_group | hash('md5') | truncate(16, True, '') + (61535 | random | string) }}"
+
+- name: create a kevyault
```

Comment:
When doing tests we want to test 3 actions when testing a scenario

* task in check mode to ensure a change is recorded but nothing actually happened
* task in normal mode (what you have here)
* task in normal mode again to test idempotency and assert that a change was not reported